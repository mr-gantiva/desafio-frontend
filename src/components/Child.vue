<script setup>
import { ref, defineProps } from 'vue'

const mensaje = ref('')

</script>

<template>
    <div class="container">
        <h2 class="text-center">Componenten hijo</h2>

        <div class="row">
            <div class="col-8 offset-2">
                <div class="container">
                    <div class="mb-3">
                        <input v-model="mensaje" type="text" class="form-control" placeholder="Escribe un mensaje">
                    </div>
                    <div class="mb-3 d-flex justify-content-center">
                        <button class="btn btn-success" @click="$emit('enviar', mensaje)">Enviar mensaje al
                            padre</button>
                    </div>
                </div>
            </div>
        </div>

    </div>
</template>

<style scoped></style>