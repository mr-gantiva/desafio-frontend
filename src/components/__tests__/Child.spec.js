import { describe, it, expect, test } from 'vitest'
import { mount, shallowMount } from '@vue/test-utils'
import Child from '../Child.vue'

describe('Componente Child', () => {
  it('Emitir un mensaje cuando se hace clic en el botón', async () => {
    const wrapper = mount(Child)
    // Encontrar el input y el botón
    const input = wrapper.find('input')
    const button = wrapper.find('button')

    // Simular la escritura en el input
    const testMensaje = 'Hola este es el mensaje de prueba'
    await input.setValue(testMensaje)

    // Simular el clic en el botón
    await button.trigger('click')

    // Verificar que se haya emitido el evento 'enviar'
    expect(wrapper.emitted()).toHaveProperty('enviar')

    // Verificar el contenido del evento emitido
    const eventoEmitido = wrapper.emitted('enviar')
    expect(eventoEmitido).toHaveLength(1)
    expect(eventoEmitido[0]).toEqual([testMensaje])
  })
  it('No emitir un mensaje si el input está vacío', async () => {
    const wrapper = mount(Child)
    // Encontrar el botón
    const button = wrapper.find('button')

    // Simular el clic en el botón
    await button.trigger('click')

    // Verificar que el evento fue emitido con un string vacio
    const eventoEmitido = wrapper.emitted('enviar')
    expect(eventoEmitido[0]).toEqual([''])
  })
  test('Snapshot de Child.vue', () => {
    // Selección del sujeto de pruebas
    const wrapper = shallowMount(Child)
    // Validación de match con el snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })
})
