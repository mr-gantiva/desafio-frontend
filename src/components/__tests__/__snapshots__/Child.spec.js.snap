// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Child > Snapshot de Child.vue 1`] = `
"<div class="container">
  <h2 class="text-center">Componenten hijo</h2>
  <div class="row">
    <div class="col-8 offset-2">
      <div class="container">
        <div class="mb-3"><input type="text" class="form-control" placeholder="Escribe un mensaje"></div>
        <div class="mb-3 d-flex justify-content-center"><button class="btn btn-success">Enviar mensaje al padre</button></div>
      </div>
    </div>
  </div>
</div>"
`;
